libs/shared/src/main/java/io/crate/common/collections/Lists2.java

```diff
@@ -101,7 +101,7 @@ public static boolean equals(Collection<?> list1, Collection<?> list2) {
      */
     public static <I, O> List<O> map(Collection<I> list, Function<? super I, ? extends O> mapper) {
         if (list.isEmpty()) {
-            return List.of();
+            return new ArrayList<>();
```

Comment:
This allows empty arrays to be mutable such that values can be added.

Reply:
We try to keep all collections by default immutable. There are exceptions to this for performance reason, but I think the core libraries should follow this rule. Where is the code part where you need to add to the list after a map operation? Can you not just create a new mutable List there ?

Reply:
I agree, we should't do this here in `Lists2` or if it turns out to be the only convenient way, we should at least have a separate method explaining this behaviour (mutable List).

Reply:
Sorry this was my mistake. While extracting `array_set` from my other PR, I carried this over by accident. Reverting these `new ArrayList<>()` should not break anything. I will keep this in mind for the remaining parts of #14140.

Reply:
👍